fv3core/stencils/c_sw.py

```diff
@@ -251,10 +251,10 @@ def circulation_cgrid(
         vort_c = fx[0, -1, 0] - fx - fy[-1, 0, 0] + fy
 
         with horizontal(region[i_start, j_start], region[i_start, j_end + 1]):
-            vort_c += fy[-1, 0, 0]
+            vort_c += dyc[-1, 0] * vc[-1, 0, 0]
 
         with horizontal(region[i_end + 1, j_start], region[i_end + 1, j_end + 1]):
-            vort_c -= fy[0, 0, 0]
+            vort_c -= dyc * vc
 
```

Comment:
This works. You could consider making the regions independent of the main domain computation, though is not necessary. e.g.
`vort_c = fx[0, -1, 0] - fx + fy` for the first region and `vort_c = fx[0, -1, 0] - fx - fy[-1, 0, 0]` for the second. This may cause validation to fail, but in this case it is justifiable to loosen the empirical error thresholds since we can see exactly the math change that would cause the answer to be slightly different. If you prefer not to mess with that, I am fine with us leaving this as-is.

Reply:
Oh that's a good idea. I'll switch it to that.